www/js/plugin/storage.ts

```diff
@@ -30,7 +30,7 @@ const localStorageSet = (key: string, value: {[k: string]: any}) => {
 
 const localStorageGet = (key: string) => {
   const value = localStorage.getItem(key);
-  if (value) {
+  if (value && value != "undefined") {
```

Comment:
I read the commit for this and I see that the string `"undefined"` was getting stored for one of the tests (not sure which one)

What I want to know is why it got stored as such in the first place. Would it be cleaner to catch this before `"undefined"` ever gets stored and prevent this from happening?

Reply:
The second test is the one that was creating that error, I'll start working through why now

Reply:
After digging through the log statements, the error is happening during the all to `markConsented`with the error `SyntaxError: "undefined" is not valid JSON`

Continuing to look at this, and I believe there is a timing issue - in `markConsented`, the first call is to `readConsent`, so that the required consent is read from the config. This, however, launches `checkNativeConsent` in the background, which sometimes times correctly to find the local consent (as it has now been marked) but not the native consent (as that is still in process). This causes an error to display (to the user) and could potentially be confusing. 

However, that is not the problem here, I'm guessing the issue is when `storageGet(DATA_COLLECTION_CONSENTED_PROTOCOL)` and the value is not defined it is for some reason finding the string "undefined" - though that makes very little sense. Maybe with the mark consented it is storing undefined?

Reply:
This error: `SyntaxError: "undefined" is not valid JSON` happens specifically the second time `readConsentState` is called, and the error is thrown during the call for `storageGet` for  `data_collection_consented_protocol` -- the first time that `storageGet` call is made, I see log statements about `uc_stored_val should be non-null` and then a message that it is copying native to local ... maybe this is where `"undefined"` get's introduced?

Reply:
This copying is happening in `getUnifiedValue` where the native value is undefined, and the local value in null -- I don't think either of them have been set yet, so I'm not sure why we're copying the native undefined over the local null, but maybe that's what we want to do?

Reply:
I've now tried checking the types throughout the process, and can't determine why the `undefined` is a string. When it is ready, it's type is undefined, and when it is set, it's type is undefined. But then when we set it, we stringify for storage. So when it is read, the type is a string.

Could I add a check to the set to only set it if it exists? Or be checking in `getUnifiedValue` for undefined alongside null?

Reply:
Added this code snippet, and all the tests now run, without the hacky string test.
```
  if (value) {
    localStorage.setItem(key, JSON.stringify(value));
  }
```